apollo-router/tests/integration/telemetry/metrics.rs

```diff
@@ -198,21 +198,21 @@ async fn test_graphql_metrics() {
     router.assert_started().await;
     router.execute_default_query().await;
     router
-        .assert_metrics_contains(r#"graphql_field_list_length_sum{graphql_field_name="topProducts",graphql_field_type="Product",graphql_type_name="Query",otel_scope_name="apollo/router"} 3"#, None)
+        .assert_metrics_contains(r#"graphql_field_list_length_sum{graphql_field_name="topProducts",graphql_field_type="Product",graphql_type_name="Product",otel_scope_name="apollo/router"} 3"#, None)
         .await;
     router
-        .assert_metrics_contains(r#"graphql_field_list_length_bucket{graphql_field_name="topProducts",graphql_field_type="Product",graphql_type_name="Query",otel_scope_name="apollo/router",le="5"} 1"#, None)
+        .assert_metrics_contains(r#"graphql_field_list_length_bucket{graphql_field_name="topProducts",graphql_field_type="Product",graphql_type_name="Product",otel_scope_name="apollo/router",le="5"} 1"#, None)
         .await;
     router
-        .assert_metrics_contains(r#"graphql_field_execution_total{graphql_field_name="name",graphql_field_type="String",graphql_type_name="Product",otel_scope_name="apollo/router"} 3"#, None)
+        .assert_metrics_contains(r#"graphql_field_execution_total{graphql_field_name="name",graphql_field_type="String",graphql_type_name="String",otel_scope_name="apollo/router"} 3"#, None)
         .await;
     router
-        .assert_metrics_contains(r#"graphql_field_execution_total{graphql_field_name="topProducts",graphql_field_type="Product",graphql_type_name="Query",otel_scope_name="apollo/router"} 1"#, None)
+        .assert_metrics_contains(r#"graphql_field_execution_total{graphql_field_name="topProducts",graphql_field_type="Product",graphql_type_name="Product",otel_scope_name="apollo/router"} 4"#, None)
```

Comment:
There are some type name changes in all of these, for which I've already posted a discussion point. This line specifically changes the count as well, which highlights the overcounting issue for array fields. Right now, the three attributes we're checking will be equivalent for a list field and all it's items, so we need to differentiate in some way.